torch/_dynamo/utils.py

```diff
@@ -1678,6 +1678,12 @@ def defake(x):
 # NB: The dictionary has to be created lazily after TorchPatcher is called so
 # that we pick up the disabled torch.utils.checkpoint wrapper. Therefore, it is
 # sitting in a separate function.
+# We also need the original untouched/ not disabled torch utils checkpoint
+# becuase distributed checkpointed wrappers import these utils before
+# TorchDynamo TorchPatcher runs.
+untouched_torch_utils_checkpoint = torch.utils.checkpoint.checkpoint
```

Comment:
i guess this is one more straw on the back of 'lets overhaul skipfiles/disable? seems pretty hard to maintain and reason about this logic.  i'm not sure if there is a better way to go forward now.

Reply:
I know this is getting out of hand. I could not come up with a cleaner way for this right now.